### PR TITLE
Updated WPK to get WPK certificates from AWS

### PR DESCRIPTION
**Pull request by okynos**
Hello team,

We have added some modifications to WPK dockerfiles in order to retrieve signing keys from AWS instead of directory (This option remains usable).
This will improve the Fargate integration.

Regards.

### CODE REVIEW

File: wpk/windows/Dockerfile [hunk 1]
```diff
@@ -4,10 +4,12 @@ RUN apt-get update && \
...
       python \
       git \
-      curl && \
+      curl \
+      jq && \
     curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
     python get-pip.py && \
-    pip install --upgrade cryptography 
+    pip install --upgrade cryptography && \
+    pip install awscli
```
Comment (BraulioV) [inline wpk/windows/Dockerfile:12]:
```suggestion
    pip install --upgrade cryptography awscli && \
```

File: wpk/run.sh [hunk 2]
```diff
@@ -61,12 +166,12 @@ main() {
 
     if [ "${NO_COMPILE}" == false ]; then
       # Execute gmake deps if the version is greater or equal to 3.5
-      if [ $MAJOR -ge 3 ] && [ $MINOR -ge 5 ]; then
+      if [ ${MAJOR} -ge 3 ]; then
```
Comment (BraulioV) [inline wpk/run.sh:169]: I prefer to keep the old check. This allows building WPK packages from older versions.

File: wpk/run.sh [hunk 3]
```diff
@@ -2,36 +2,141 @@
...
+
+
     NO_COMPILE=false
     # Get Wazuh
-    git clone $REPOSITORY $DIRECTORY || exit 1
-    cd $DIRECTORY
-    git checkout $BRANCH
+    git clone ${REPOSITORY} ${DIRECTORY} || exit 1
+    cd ${DIRECTORY}
+    git checkout ${REFERENCE}
```
Comment (BraulioV) [inline wpk/run.sh:139]:
```suggestion
    git clone ${REPOSITORY} ${DIRECTORY} -b ${REFERENCE} --depth=1 || exit 1
    cd ${DIRECTORY}
```

File: wpk/run.sh [hunk 4]
```diff
@@ -2,36 +2,141 @@
...
+                shift 2
+            fi
+          ;;
+        "-h"|"--help")
+            help 0
+            ;;
+        *)
+            help 1
+        esac
+    done
```
Comment (BraulioV) [inline wpk/run.sh:122]:
```suggestion
    while [ -n "${1}" ]
    do
        case "${1}" in
            "-b"|"--branch")
                if [ -n "${2}" ]; then
                    REFERENCE="$(echo ${2} | cut -d'/' -f2)"
                    shift 2
                else
                    echo "ERROR: Missing branch."
                    help 1
                fi
                ;;
            "-o"|"--output")
                if [ -n "${2}" ]; then
                    OUT_NAME="${2}"
                    shift 2
                else
                    echo "ERROR: Missing output name."
                    help 1
                fi
                ;;
            "-r"|"--revision")
                if [ -n "${2}" ]; then
                    REVISION="${2}"
                    shift 2
                fi
                ;;
            "-p"|"--path")
                if [ -n "${2}" ]; then
                    INSTALLATION_PATH="${2}"
                    shift 2
                fi
                ;;
            "-pn"|"--package-name")
                if [ -n "${2}" ]; then
                    PKG_NAME="${2}"
                    HAVE_PKG_NAME=true
                    shift 2
                fi
                ;;
            "-j"|"--jobs")
                if [ -n "${2}" ]; then
                    JOBS="${2}"
                    shift 2
                fi
                ;;
            "-c"|"--checksum")
                CHECKSUM="yes"
                shift 1
                ;;
            "-wk"|"--wpk-key")
                if [ -n "${2}" ]; then
                    AWS_WPK_KEY="${2}"
                    shift 2
                fi
                ;;
            "-wc"|"--wpk-cert")
                if [ -n "${2}" ]; then
                    AWS_WPK_CERT="${2}"
                    shift 2
                fi
                ;;
            "-wr"|"--wpk-key-region")
                if [ -n "${2}" ]; then
                    AWS_REGION="${2}"
                    shift 2
                fi
            ;;
            "-h"|"--help")
                help 0
                ;;
            *)
                help 1
        esac
    done
```

File: wpk/run.sh [hunk 5]
```diff
@@ -2,36 +2,141 @@
...
-JOBS=${2}
-OUT_NAME=${3}
-CHECKSUM=${4}
-INSTALLATION_PATH=${5}
-PKG_NAME=${6}
+REFERENCE=""
+JOBS="4"
+OUT_NAME=""
+CHECKSUM="no"
+INSTALLATION_PATH="/var"
```
Comment (BraulioV) [inline wpk/run.sh:9]: Can it be changed to  `/var/ossec`?

File: wpk/generate_wpk_package.sh [hunk 6]
```diff
@@ -72,22 +97,25 @@ function build_container() {
...
-  echo "    -h,   --help                                Show this help."
+  echo "    -t,   --target-system <target> [Required] Select target wpk to build [linux/windows]"
+  echo "    -b,   --branch <branch>        [Required] Select Git branch or tag e.g. $BRANCH"
+  echo "    -d,   --destination <path>     [Required] Set the destination path of package."
+  echo "    -pn,  --package-name <name>    [Required for windows] Package name to pack on wpk."
+  echo "    -o,   --output <name>          [Required] Name to the output package."
+  echo "    -k,   --key-dir <arch>         [Optional] Set the WPK key path to sign package."
+  echo "    -wk,  --wpk-key                [Optional] AWS Secrets manager Name/ARN to get WPK private key."
+  echo "    -wc,  --wpk-cert               [Optional] AWS secrets manager Name/ARN to get WPK certificate."
+  echo "    -wr,  --wpk-key-region         [Optional] AWS Region where secrets are stored."
```
Comment (BraulioV) [inline wpk/generate_wpk_package.sh:113]:
```suggestion
  echo "    --aws-wpk-key                [Optional] AWS Secrets manager Name/ARN to get WPK private key."
  echo "    --aws-wpk-cert               [Optional] AWS secrets manager Name/ARN to get WPK certificate."
  echo "    --aws-wpk-key-region         [Optional] AWS Region where secrets are stored."
```

File: wpk/windows/Dockerfile (hunk 1 above)
Comment (okynos) [reply wpk/windows/Dockerfile]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

File: wpk/run.sh (hunk 2 above)
Comment (okynos) [reply wpk/run.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

File: wpk/run.sh (hunk 3 above)
Comment (okynos) [reply wpk/run.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

File: wpk/run.sh (hunk 4 above)
Comment (okynos) [reply wpk/run.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

File: wpk/run.sh (hunk 5 above)
Comment (okynos) [reply wpk/run.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

File: wpk/generate_wpk_package.sh (hunk 6 above)
Comment (okynos) [reply wpk/generate_wpk_package.sh]: Done in https://github.com/wazuh/wazuh-packages/pull/467/commits/1c82879d25119ef9aeb3808e1a4f12747d8edbf0

**Pull request merged**